hal/src/main/native/athena/HAL.cpp

```diff
@@ -270,6 +273,23 @@ int64_t HAL_GetFPGARevision(int32_t* status) {
   return global->readRevision(status);
 }
 
+size_t HAL_GetSerialNumber(char* buffer, size_t size) {
+  const char* serialNum = std::getenv("serialnum");
+  if (serialNum) {
+    size_t length = std::strlen(serialNum);
+    std::strncpy(buffer, serialNum, size);
```

Comment:
This isn't fully safe, because if length == size the string will not be null terminated. 

TBH, since this is a rarely used call, I'd probably prefer to just allocate here, that way we don't have to worry about it at all. You can see HAL_GetJoystickName and HAL_FreeJoystickName for an example on how to do that method properly.

Reply:
Actually nevermind on the 2nd part. That complicated the sim side. So we just need to fix the null terminated part.

Reply:
Does the returned string *need* to be null terminated? The function returns a size and our functions that call that function use the size.

Reply:
It we're gonna do that, then we shouldn't be null terminating anywhere. Which would mean using memcpy instead of strncpy, all the other places that either add a null terminator or check for them get removed, and then using an 8 length buffer. Either null terminated everywhere, or null terminated nowhere.